src/openrct2-ui/windows/TrackList.cpp

```diff
@@ -337,7 +337,10 @@ static void window_track_list_mouseup(rct_window* w, rct_widgetindex widgetIndex
             // Keep the highlighted item selected
             if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
             {
-                w->selected_list_item = _filteredTrackIds[w->selected_list_item];
+                if (static_cast<ssize_t>(_filteredTrackIds.size() - 1) >= w->selected_list_item)
```

Comment:
Why is there a subtraction of 1? I may be missing something, but it looks like `_filteredTrackIds.size() > w->selected_list_item` would be sufficient here.